scrowl-io/scrowl
# [sc-191] - Setup VSCode debugging

### Short summary

- Remove forge congif from package.json into its own js file
   - This lets us read in and use env vars
   - This was needed to avid port conflicts between the app and the debug instance
- Add launch.json with config to be able to use the built in debug tools in VS Code both the main and render processes. 
- Add new env vars to be used by the `forge.config.js` file
- Started documentation (its empty atm) for how to use it
- Instructions added into the repo:
   - https://github.com/EEBOS/SCROWL/wiki/4.-Debug-With-VS-Code
   - But much better with gifs here: https://www.notion.so/Debugging-in-VSCode-63dccb0d36f34676aa901eb8d3540e1a
---

### Test steps

1. while in root of the project
   1. `make start startOver=true` 
1. Once the project is running, attach a remote container session
   1. Go to the `Docker` extension, then right click on `Scrowl_electron` container
   2. Select `Attach Visual Studio Code`
   3. ![image](https://user-images.githubusercontent.com/93738190/153439223-ed5ef784-8903-4d99-82cc-fe24877f144e.png)
1. A new VS Code instance should pop up
2. Click on the debug menu item on the right
3. Click on the green play icon to start a debug session on the `Electron:Main`
   1. ![image](https://user-images.githubusercontent.com/93738190/153439811-a97bd36b-f551-4c27-b3a0-4675ef448791.png)
   2. You can also just press `f5` 
2. You should see another instance of Scrowl running.
3. At this point you can add brake points in the `index.ts` (will be renamed to main probably) file and restart the debug session by press `f5`

There will be more documentation coming. It will be in the repo's wiki first.
